### PR TITLE
Added RegisterServices option to Install-ContainerTools function

### PR DESCRIPTION
**Pull request by billywr**
Adding RegisterServices option in the Install-ContainerTools function introduces the ability to register and start containerd and buildkit services.

when you run Install-ContainerTools -RegisterServices above mentioned services should get registered and start.
image

### CODE REVIEW

File: docs/command-reference.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ Downloads container tool (Containerd, BuildKit, nerdctl) asynchronously and inst
 | DownloadPath | String | Path to download container tools  | `$HOME\Downloads` |
 | Cleanup | Switch | Specifies whether to cleanup after installation is done  |  |
 | Force | Switch | Force install the tools even if they already exists at the specified path |  |
+| Setup | Switch | Register and Start Conatinerd and Buildkitd services |  |
```
Comment (profnandaa) [inline docs/command-reference.md:65]: nit: I feel like the name `Setup` is too generic, perhaps something more idiomatic like `RegisterService` would be better?

Comment (profnandaa) [reply docs/command-reference.md:65]: Though also thinking about it, could we just have more of opt-out than opt-in? That is, by default, register the service but if someone wants to opt-out, they can provide the opt-out switch like `SkipServiceRegistration` or something.

**Review by profnandaa — COMMENTED** [review]
Other than, the rest LGTM.

**Comment by profnandaa** [comment]
@TinaMor -- can complete setting up the `ctk-ci` pipeline?

File: docs/command-reference.md (hunk 1 above)
Comment (iankingori) [reply docs/command-reference.md:65]: @billyoo I had initially proposed `Register-Service`/`Register-Services` as the switch. I think we can make that change then make a decision on what the default should be after.

Comment (TinaMor) [reply docs/command-reference.md:65]: @billyoo `Register and Start Conatinerd and Buildkitd services and set up NAT network`

Comment (TinaMor) [reply docs/command-reference.md:65]: @iankingori `RegisterService`. The hyphen is not a vaid param name.

File: containers-toolkit/Public/AllToolsUtilities.psm1 [hunk 2]
```diff
@@ -153,12 +157,30 @@ function Install-ContainerTools {
...
+                    }
                 }
                 catch {
                     Write-Error "Installation failed for $($params.feature). $_"
                 }
             }
 
+            if ($RegisterServices) {
+                Initialize-NatNetwork
+            }else {
```
Comment (TinaMor) [inline containers-toolkit/Public/AllToolsUtilities.psm1:177]: Liniting.

File: containers-toolkit/Public/AllToolsUtilities.psm1 [hunk 3]
```diff
@@ -68,7 +68,11 @@ function Install-ContainerTools {
 
         [Switch]
         [parameter(HelpMessage = "Force install the tools even if they already exists at the specified path")]
-        $Force
+        $Force,
+
+        [switch]
+        [parameter(HelpMessage = "Register and Start Conatinerd and Buildkitd services")]
```
Comment (TinaMor) [inline containers-toolkit/Public/AllToolsUtilities.psm1:74]: .., and set up NAT network

**Review by TinaMor — COMMENTED** [review]
Minor change requests. LGTM

File: containers-toolkit/Public/AllToolsUtilities.psm1 [hunk 4]
```diff
@@ -153,12 +157,30 @@ function Install-ContainerTools {
                     Install-RequiredFeature @InstallParams -Cleanup $CleanUp
 
                     $completedInstalls += $params.Feature
+
+                    if ($RegisterServices) {
```
Comment (TinaMor) [inline containers-toolkit/Public/AllToolsUtilities.psm1:161]: Any tests for this new additions?

**Review by iankingori — APPROVED** [review]
LGTM

File: Tests/AllToolsUtilities.Tests.ps1 [hunk 5]
```diff
@@ -63,6 +63,9 @@ Describe "AllToolsUtilities.psm1" {
             Mock Install-ContainerToolConsent -ModuleName 'AllToolsUtilities' -MockWith { return $true }
             Mock Install-ContainerTools -ModuleName 'AllToolsUtilities'
             Mock Test-EmptyDirectory  -ModuleName 'BuildkitTools' -MockWith { return $true }
```
Comment (TinaMor) [inline Tests/AllToolsUtilities.Tests.ps1:65]: Please change `BuildkitTools` to `AllToolsUtilities`. This method `Test-EmptyDirectory` may not even be called in this script. We should consider removing it altogether

File: Tests/AllToolsUtilities.Tests.ps1 [hunk 6]
```diff
@@ -202,5 +205,23 @@ Describe "AllToolsUtilities.psm1" {
...
+            -RegisterServices
+
+            $RegisterContainerdParams = @{
+                ContainerdPath =  "$InstallPath\Containerd"
+            }
+            $RegisterBuildKitdParams = @{
+                BuildKitPath = "$InstallPath\Buildkit"
+            }
+
+            Should -Invoke Register-ContainerdService -ModuleName 'AllToolsUtilities' -Times 1 -ParameterFilter { $RegisterContainerdParams -and $Start -eq $true -and $Force -eq $true }
```
Comment (TinaMor) [inline Tests/AllToolsUtilities.Tests.ps1:222]: Nit: Linting long lines

Comment (TinaMor) [inline Tests/AllToolsUtilities.Tests.ps1:222]: In the test `"Should use defaults"` (line 84), we should test that these methods are not called when the `RegisterServices` parameter is not passed

**Review by TinaMor — APPROVED** [review]

**Review by profnandaa — APPROVED** [review]

**Review by CharityKathure — APPROVED** [review]
LGTM

**Pull request merged**